### PR TITLE
DucklingHTTPExtractor with defined request timeout

### PR DESCRIPTION
**Pull request by vishnuvrpriya**
**Proposed changes**:
- Fixed the bug being discussed under #1695 
- Added a timeout of a default of 3 seconds to t.he Duckling HTTP Request Invocation and made it a configurable component parameter.
- Updated the docs to reflect the new timeout parameter.
- Ran a few tests to ascertain the fix.Attached a screenshot (below) of the test that fulfills the fix.
DucklingHTTPTimeout
 

**Status (please check what you already did)**:
- [x] made PR ready for code review 
- [ ] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

**Comment by msamogh** [comment]
Thanks for the PR, we'll give it a review as soon as possible.

File: rasa/nlu/extractors/duckling_http_extractor.py [hunk 1]
```diff
@@ -89,6 +92,10 @@ def _locale(self):
             self.component_config["locale"] = locale_fix
         return self.component_config.get("locale")
 
+
+    def _timeout(self):
```
Comment (imLew) [inline rasa/nlu/extractors/duckling_http_extractor.py]: I don't think it's necessary to create a method for this, as it' only getting the value from the dict.

File: rasa/nlu/extractors/duckling_http_extractor.py [hunk 2]
```diff
@@ -113,7 +120,7 @@ def _duckling_parse(self, text, reference_time):
                 "Content-Type": "application/x-www-form-urlencoded; charset=UTF-8"
             }
             response = requests.post(
-                self._url() + "/parse", data=payload, headers=headers
+                self._url() + "/parse", data=payload, headers=headers,timeout=self._timeout()
```
Comment (imLew) [inline rasa/nlu/extractors/duckling_http_extractor.py]:
```suggestion
                self._url() + "/parse", data=payload, headers=headers, timeout=self.component_config.get("timeout")
```

You can simply get the value from the dict here.

**Comment by vishnuvrpriya** [comment]
Hi @imLew ,Thanks for the valuable comments.I will address them and get back to you!

File: rasa/nlu/extractors/duckling_http_extractor.py (hunk 1 above)
Comment (vishnuvrpriya) [reply rasa/nlu/extractors/duckling_http_extractor.py]: Addressed this review comment.

File: rasa/nlu/extractors/duckling_http_extractor.py (hunk 2 above)
Comment (vishnuvrpriya) [reply rasa/nlu/extractors/duckling_http_extractor.py]: Addressed this review comment.

**Comment by vishnuvrpriya** [comment]
Hi @imLew , I have addressed the review comments that you suggested.I have also formatted the code using black and have successfully finished signing the CLA.

**Comment by imLew** [comment]
@vishnuvrpriya thanks!
Just one more thing: it seems like github does not know that your local git user is the same as your github user, that's why there are two names that needs to sign the CLA. Unfortunately we cannot merge this PR unless all contributors have signed the CLA.
I've done bit of searching and maybe the answers to https://webapps.stackexchange.com/questions/20309/how-to-claim-contributions-on-github can help you claim the commits you made locally.

**Comment by vishnuvrpriya** [comment]
Hi @imLew . I am unable to solve the CLA issue and so I am trying to close and reopen the PR. Please bear with me for the inconvenience.

**Comment by vishnuvrpriya** [comment]
Hi @imLew , I am unable to solve the CLA issue and I am trying to close and reopen the PR.Please bear with me for the inconvenience.

**Comment by imLew** [comment]
@vishnuvrpriya am I right in assuming the git user that did not sign the CLA is your local user.name?
If so, does it have an email address associated with it? If yes you should be able to add that email address to your github account. Once it's added the CLA should either be signed or you should be able to sign it both users.
If not I'm not sure if you can change this PR. You might however be able to somehow change it locally (e.g. adding an email address to it) and create another PR.

**Comment by imLew** [comment]
@vishnuvrpriya, I asked around and it seems like you will have to create a new PR if you cannot add the email to your github account.
To get the new PR to associate the commits with your github account you can do a soft reset or you can try https://github.com/jayphelps/git-blame-someone-else (note that I haven't tried either of these methods myself). Before that of course make sure that your local git users email address is the same as the one linked to your github account.

**Comment by vishnupriyavr** [comment]
Hi @imLew ,Thanks for the guidance.I actually had to create a new user and a new PR #4535 to successfully complete the CLA.I request you to review the PR and help in merging it.I will also be closing this PR to avoid any confusion.

**Pull request closed without merging**